README.md

```diff
@@ -726,3 +730,57 @@ By default, the client classes will automatically retry the following errors up
 - 502 Bad Gateway
 - 503 Service Unavailable
 - 504 Gateway Timeout
+
+## Observability with OpenTelemetry
+
+> Note:
+> OpenAI .NET SDK instrumentation is in development and is not complete. See [Available sources and meters](#available-sources-and-meters) section for the list of covered operations.
+
+OpenAI .NET library is instrumented with distributed tracing and metrics using .NET [tracing](https://learn.microsoft.com/dotnet/core/diagnostics/distributed-tracing)
+and [metrics](https://learn.microsoft.com/dotnet/core/diagnostics/metrics-instrumentation) API and supports [OpenTelemetry](https://learn.microsoft.com/dotnet/core/diagnostics/observability-with-otel).
+
+OpenAI .NET library follows [OpenTelemetry Semantic Conventions for Generative AI systems](https://github.com/open-telemetry/semantic-conventions/tree/main/docs/gen-ai).
+
+### How to enable
+
+The instrumentation is **experimental** - names of activity sources and meters, volume and semantics of the telemetry items may change.
+
+To enable the instrumentation:
+
+1. Set instrumentation feature-flag using one of the following options:
+
+   - set the `AZURE_EXPERIMENTAL_ENABLE_ACTIVITY_SOURCE` environment variable to `"true"`
+   - set the `Azure.Experimental.EnableActivitySource` context switch to true in your application code when application
+     is starting and before initializing any OpenAI clients. For example:
+
+     ```csharp
+     AppContext.SetSwitch("Azure.Experimental.EnableActivitySource", true);
+     ```
+
+2. Configuring OpenTelemetry to record telemetry from OpenAI sources and meters:
+
+   ```csharp
+   builder.Services.AddOpenTelemetry()
+       .WithTracing(b =>
+       {
+           b.AddSource("OpenAI.*")
+             ...
+            .AddOtlpExporter();
+       })
+       .WithMetrics(b =>
+       {
+           b.AddMeter("OpenAI.*")
+            ...
+            .AddOtlpExporter();
+       });
+   ```
+
+Consider enabling [HTTP client instrumentation](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.Http) to see all HTTP client
+calls made by your application including those done by the OpenAI SDK.
+Check out [OpenTelemetry documentation](https://opentelemetry.io/docs/languages/net/getting-started/) for more details.
+
+### Available sources and meters
```

Comment:
it might not be obvious what are "sources" and "meters"..

Reply:
Agreed -- this bears a bit more of a concept-level introduction. We've already heard feedback that the top-level README is very big, though, so just adding more seems to exacerbate that; perhaps we can use this topic as a pilot for separating detailed topic information into a sub-page? E.g., as a pattern-match, check out how the Python SDK splits out streaming helpers: https://github.com/openai/openai-python?tab=readme-ov-file#streaming-helpers

I'd propose that we put this in a `/docs` folder rather than repo root, but we could then put a dedicated header section into the top-level README for instrumentation explaining what it is and why you'd use it, then a link to the `/docs/instrumentation.md` for follow-up.

Reply:
done, PTAL